srml/staking/src/lib.rs

```diff
@@ -1380,3 +1380,28 @@ impl<T: Trait> SelectInitialValidators<T::AccountId> for Module<T> {
 		<Module<T>>::select_validators().1
 	}
 }
+
+/// A type for performing slashing based on severity of misbehavior
+pub struct StakingSlasher<T>(rstd::marker::PhantomData<T>);
```

Comment:
Just to understand things better, where is this used exactly?

Reply:
it is not used yet but it would be used as follows:

1. Consensus Module (reports violation)
2. After that the violation is proofed, calls `ReportSlash:slash` (this will differ between different classes of slashing)
3. ReportSlash::slash calls `StakingSlasher::slash` and this performs the actual slashing and rewarding